### PR TITLE
NPE in NewlinesRule:

### LINKED ISSUE #1047: NPE in NewlinesRule
Consider the following code:
```kotlin
fun foo() {
    val a = it.responseBody!![0].name
}
```
NPE:

```bash
[ERROR] 2021-09-06 18:22:12 Internal error has occurred in rule [newlines].
Please make an issue on this bug at https://github.com/cqfn/diKTat/. 
As a workaround you can disable these inspections in yml config: <[COMPLEX_EXPRESSION, REDUNDANT_SEMICOLON, WRONG_NEWLINES]>.
java.lang.NullPointerException: astNode.treePrev must not be null 
at org.cqfn.diktat.ruleset.rules.chapter3.files.NewlinesRule.isNotValidCalls(NewlinesRule.kt:557)               
at org.cqfn.diktat.ruleset.rules.chapter3.files.NewlinesRule.isCallsChain(NewlinesRule.kt:532)
at org.cqfn.diktat.ruleset.rules.chapter3.files.NewlinesRule.handleOperatorWithLineBreakBefore(NewlinesRule.kt:186)         
at org.cqfn.diktat.ruleset.rules.chapter3.files.NewlinesRule.logic(NewlinesRule.kt:126)
at org.cqfn.diktat.ruleset.rules.DiktatRule.visit(DiktatRule.kt:50)
at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:271)
at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:23)                  
```

### PR DESCRIPTION
**Pull request by Cheshiriks**
### What's done:
* fixed bug in NewlinesRule
Closes #1047

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt [hunk 1]
```diff
@@ -461,7 +462,7 @@ class NewlinesRule(configRules: List<RulesConfig>) : DiktatRule(
 
         if (psi.children.isNotEmpty() && !psi.isFirstChildElementType(DOT_QUALIFIED_EXPRESSION) &&
                 !psi.isFirstChildElementType(SAFE_ACCESS_EXPRESSION)) {
-            val firstChild = psi.firstChild
+            val firstChild = if (psi.isFirstChildElementType(ARRAY_ACCESS_EXPRESSION)) psi.firstChild.firstChild else psi.firstChild
```
Comment (kgevorkyan) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt:465]: Is that just workaround?

I looked at this bug before and think, that the main problem is in these lines:

https://github.com/cqfn/diKTat/blob/37b2478924f69d9f4be205bc46a2a12bbed89fb8/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt#L469-L471

For now I didn't catch the logic of this `if` statement - in which cases do we need this, even more it doesn't covered by any test.
Probably, we should modify it or get rid of it at all.

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/files/NewlinesRuleWarnTest.kt [hunk 2]
```diff
@@ -527,6 +527,18 @@ class NewlinesRuleWarnTest : LintTestBase(::NewlinesRule) {
         )
     }
 
+    @Test
+    @Tag(WarningNames.WRONG_NEWLINES)
+    fun `should warn for array access expression`() {
+        lintMethod(
+            """
+                    |fun bar(): String {
+                    |    val a = it.responseBody!!.name
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/files/NewlinesRuleWarnTest.kt:536]: Yes. I removed this for a test and forgot to return

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt (hunk 1 above)
Comment (Cheshiriks) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt:465]: Yes, I deleted this piece and. everything continued to work

Comment (kgevorkyan) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt:465]: Yes, it should work, however did you understand the reason, why did we need this before, for which cases?

I can't imagine any other examples for now, which will satisfy this condition, instead of `it.responseBody!![0].name`, for which it doesn't work

Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt:465]: Don't get how it should fix anything?

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/files/NewlinesRuleWarnTest.kt [hunk 3]
```diff
@@ -527,6 +527,18 @@ class NewlinesRuleWarnTest : LintTestBase(::NewlinesRule) {
         )
     }
 
+    @Test
```
Comment (akuleshov7) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/files/NewlinesRuleWarnTest.kt:530]: create fixer tests (where `WRONG_NEWLINES` is fixing the code in such case) to prove that you are right

**Review by akuleshov7 — CHANGES_REQUESTED** [review]
add more tests, make the logic at least look less hardcoded (create null-checks and describe your workarounds)

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/files/NewlinesRuleWarnTest.kt [hunk 4]
```diff
@@ -527,6 +527,18 @@ class NewlinesRuleWarnTest : LintTestBase(::NewlinesRule) {
...
+    @Tag(WarningNames.WRONG_NEWLINES)
+    fun `should warn for array access expression`() {
+        lintMethod(
+            """
+                    |fun bar(): String {
+                    |    val a = it.responseBody!![0].name
+                    |}
```
Comment (akuleshov7) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/files/NewlinesRuleWarnTest.kt:537]: please add positive scenarios (that will trigger warnings)

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/files/NewlinesRuleWarnTest.kt (hunk 3 above)
Comment (akuleshov7) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/files/NewlinesRuleWarnTest.kt:530]: @Cheshiriks where are tests for fixers?

**Review by akuleshov7 — APPROVED** [review]
@Cheshiriks
